app/kuma-dp/pkg/dataplane/envoy/remote_bootstrap.go

```diff
@@ -24,9 +24,8 @@ type remoteBootstrap struct {
 	client *http.Client
 }
 
-func NewRemoteBootstrapGenerator(client *http.Client) BootstrapConfigFactoryFunc {
-	rb := remoteBootstrap{client: client}
-	return rb.Generate
+func NewRemoteBootstrapGenerator(client *http.Client) BootstrapConfigGenerator {
+	return &remoteBootstrap{client: client}
 }
```

Comment:
```suggestion
type RemoteBootstrapGenerator {
        Client *http.Client // use DefaultClient if not set
        ControlPlane *kumadp.ControlPlane
}
```

We don't really need a `New` constructor IMHO, since there's no private fields that require initialization. We can set all the config that `Generate` needs right here.

Reply:
Ugh, I guess this needs the full config, because we have to internally transform `BootstrapParams` into `BootstrapRequest` for no apparent reason ...